datatypes/src/raster/raster.rs

```diff
@@ -0,0 +1,274 @@
+}
+
+pub trait GridPixelAccessMut<T, I> {
+    /// Sets the value at a pixels location
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geoengine_datatypes::raster::{Raster, Dim, SimpleRaster2d, TimeInterval, GridPixelAccessMut};
+    ///
+    /// let mut raster2d = SimpleRaster2d::new(
+    ///    [3, 2].into(),
+    ///    vec![1,2,3,4,5,6],
+    ///    None,
+    ///    TimeInterval::default(),
+    ///    [1.0, 1.0, 0.0, 1.0, 0.0, 1.0].into(),
+    /// ).unwrap();
+    /// raster2d.set_pixel_value_at_grid_index(&(1, 1), 9).unwrap();
+    /// assert_eq!(raster2d.data_container(), &[1,2,3,9,5,6]);
+    /// ```
+    ///
+    fn set_pixel_value_at_grid_index(&mut self, grid_index: &I, value: T) -> Result<()>;
+}
+
+pub trait CoordinatePixelAccess<T> {
+    fn pixel_value_at_coord(&self, coordinate: (f64, f64)) -> T;
+}
+
+impl<D, T, C, I> GridPixelAccess<T, I> for BaseRaster<D, T, C>
+where
+    D: GridDimension,
+    I: GridIndex<D>,
+    C: AsRef<[T]> + Capacity,
+    T: Copy,
+{
+    fn pixel_value_at_grid_index(&self, grid_index: &I) -> Result<T> {
+        let index = grid_index.grid_index_to_1d_index(&self.grid_dimension)?;
+        Ok(self.data_container.as_ref()[index])
+    }
+}
+
+impl<D, T, C, I> GridPixelAccessMut<T, I> for BaseRaster<D, T, C>
+where
+    D: GridDimension,
+    I: GridIndex<D>,
+    C: AsMut<[T]> + Capacity,
+    T: Copy,
+{
+    fn set_pixel_value_at_grid_index(&mut self, grid_index: &I, value: T) -> Result<()> {
+        let index = grid_index.grid_index_to_1d_index(&self.grid_dimension)?;
+        self.data_container.as_mut()[index] = value;
+        Ok(())
+    }
+}
+
+pub type SimpleRaster2d<T> = BaseRaster<Dim<[usize; 2]>, T, Vec<T>>;
+pub type SimpleRaster3d<T> = BaseRaster<Dim<[usize; 3]>, T, Vec<T>>;
+
+#[cfg(test)]
+mod tests {
+    use super::{Dim, GridPixelAccess, GridPixelAccessMut, SimpleRaster2d, TimeInterval};
+
+    #[test]
+    fn simple_raster_2d() {
+        let dim = [3, 2];
+        let data = vec![1, 2, 3, 4, 5, 6];
+        let geo_transform = [1.0, 1.0, 0.0, 1.0, 0.0, 1.0];
+        let temporal_bounds: TimeInterval = TimeInterval::default();
+        SimpleRaster2d::new(
+            dim.into(),
+            data,
+            None,
+            temporal_bounds,
+            geo_transform.into(),
+        )
+        .unwrap();
+    }
+
+    #[test]
+    fn simple_raster_2d_at_tuple() {
+        let tuple_index = (1, 1);
+
+        let dim = [3, 2];
+        let data = vec![1, 2, 3, 4, 5, 6];
+        let geo_transform = [1.0, 1.0, 0.0, 1.0, 0.0, 1.0];
+        let temporal_bounds: TimeInterval = TimeInterval::default();
+        let raster2d = SimpleRaster2d::new(
+            dim.into(),
+            data,
+            None,
+            temporal_bounds,
+            geo_transform.into(),
+        )
+        .unwrap();
+        let value = raster2d.pixel_value_at_grid_index(&tuple_index).unwrap();
+        assert!(value == 4);
```

Comment:
Just as a general suggestion, why don't you just use
```suggestion
        assert_eq!(
            raster2d.pixel_value_at_grid_index(&tuple_index).unwrap(),
            4
        );
```